From jejb@kernel.org  Tue Sep 30 14:36:15 2008
From: Kirill A. Shutemov <kirill@shutemov.name>
Date: Tue, 23 Sep 2008 17:25:04 GMT
Subject: smb.h: do not include linux/time.h in userspace
To: jejb@kernel.org, stable@kernel.org
Message-ID: <200809231725.m8NHP4kL022086@hera.kernel.org>

From: Kirill A. Shutemov <kirill@shutemov.name>

commit c32a162fd420fe8dfb049db941b2438061047fcc upstream

linux/time.h conflicts with time.h from glibc

It breaks building smbmount from samba.  It's regression introduced by
commit 76308da (" smb.h: uses struct timespec but didn't include
linux/time.h").

Signed-off-by: Kirill A. Shutemov <kirill@shutemov.name>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/smb.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/smb.h
+++ b/include/linux/smb.h
@@ -11,7 +11,9 @@
 
 #include <linux/types.h>
 #include <linux/magic.h>
+#ifdef __KERNEL__
 #include <linux/time.h>
+#endif
 
 enum smb_protocol { 
 	SMB_PROTOCOL_NONE, 
